### PR TITLE
DBZ-2808 Add Mongo DB to UI backend

### PR DESCRIPTION
**Pull request by rk3rn3r**
see: https://issues.redhat.com/browse/DBZ-2808

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -14,6 +14,15 @@
     <logmanager.class>org.jboss.logmanager.LogManager</logmanager.class>
   </properties>
 
+  <repositories>
+    <repository>
+      <id>sonatype-oss-snapshots</id>
+      <name>Sonatype OSS SNAPSHOT Repository</name>
+      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
+      <!--// temporary repository in order to use Debezium SNAPSHOT nightly builds //-->
```
Comment (rk3rn3r) [inline backend/pom.xml:22]: Only temporary usage to get latest changes easily for the UI team without the need to build jars by themselves.

File: backend/src/main/java/io/debezium/configserver/service/mongodb/MongoDbConnectorIntegrator.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            collections = determineDataCollectionsToBeSnapshotted(config, primary(context).collections()).collect(Collectors.toList());
+        } catch (Throwable throwable) {
+            throw new DebeziumException(throwable);
+        }
+
+        List<DataCollection> matchingTables = collections.stream()
+                .map(collectionId -> new DataCollection(collectionId.replicaSetName() + "." + collectionId.dbName(), collectionId.name()))
+                .collect(Collectors.toList());
+
+        return FilterValidationResult.valid(matchingTables);
```
Comment (rk3rn3r) [inline backend/src/main/java/io/debezium/configserver/service/mongodb/MongoDbConnectorIntegrator.java:142]: I still need to validate that part in the UI.

File: pom.xml [hunk 3]
```diff
@@ -16,25 +16,22 @@
...
-    <quarkus.plugin.version>1.8.1.Final</quarkus.plugin.version>
+    <maven.surefire.plugin.version>3.0.0-M5</maven.surefire.plugin.version>
+    <quarkus.plugin.version>1.10.2.Final</quarkus.plugin.version>
     <quarkus.platform.artifact-id>quarkus-universe-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-    <quarkus.platform.version>1.8.1.Final</quarkus.platform.version>
+    <quarkus.platform.version>1.10.2.Final</quarkus.platform.version>
     <testcontainers.version>1.14.3</testcontainers.version>
-    <debezium.base.version>1.3.0.Beta2</debezium.base.version>
+    <debezium.base.version>1.4.0-SNAPSHOT</debezium.base.version>
```
Comment (rk3rn3r) [inline pom.xml:34]: this is what we need the Sonatype OSS repo for temporary.

File: backend/src/main/java/io/debezium/configserver/service/mongodb/MongoDbConnectorIntegrator.java (hunk 2 above)
Comment (rk3rn3r) [reply backend/src/main/java/io/debezium/configserver/service/mongodb/MongoDbConnectorIntegrator.java:142]: waiting for https://github.com/debezium/debezium/pull/1999

**Comment by rk3rn3r** [comment]
@gunnarmorling I think this PR is ready for review

File: backend/pom.xml (hunk 1 above)
Comment (gunnarmorling) [reply backend/pom.xml:22]: Is this still needed at this point?

File: pom.xml (hunk 3 above)
Comment (gunnarmorling) [reply pom.xml:34]: Shouldn't be needed any longer?

**Review by gunnarmorling — APPROVED** [review]
Two comments inline. Thx!

File: backend/pom.xml (hunk 1 above)
Comment (rk3rn3r) [reply backend/pom.xml:22]: I think as some time passed now :clock1030:, I have to check but quite certainly I can revert this now.

File: pom.xml [hunk 4]
```diff
@@ -16,25 +16,22 @@
...
-    <quarkus.plugin.version>1.8.1.Final</quarkus.plugin.version>
+    <maven.surefire.plugin.version>3.0.0-M5</maven.surefire.plugin.version>
+    <quarkus.plugin.version>1.10.2.Final</quarkus.plugin.version>
     <quarkus.platform.artifact-id>quarkus-universe-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-    <quarkus.platform.version>1.8.1.Final</quarkus.platform.version>
+    <quarkus.platform.version>1.10.2.Final</quarkus.platform.version>
     <testcontainers.version>1.14.3</testcontainers.version>
-    <debezium.base.version>1.3.0.Beta2</debezium.base.version>
+    <debezium.base.version>1.4.0.CR1</debezium.base.version>
```
Comment (rk3rn3r) [inline pom.xml:34]: We just decided we'll wait for tomorrow's v1.4.0.Final release and use that released version here.

**Pull request merged**